src/unstract/adapters/vectordb/pinecone/src/pinecone.py

```diff
@@ -62,65 +64,101 @@ def get_json_schema() -> str:
+        )
+        self._collection_name = collection_name.replace("_", "-").lower()
+        dimension = self._config.get(
+            VectorDbConstants.EMBEDDING_DIMENSION,
+            VectorDbConstants.DEFAULT_EMBEDDING_SIZE,
+        )
+
+        specification = self._config.get(Constants.SPECIFICATION)
+        if specification == Constants.SPEC_POD:
+            environment = self._config.get(Constants.ENVIRONMENT)
+            spec = PodSpec(
+                environment=environment,
+                replicas=Constants.DEFAULT_SPEC_COUNT_VALUE,
+                shards=Constants.DEFAULT_SPEC_COUNT_VALUE,
+                pods=Constants.DEFAULT_SPEC_COUNT_VALUE,
+                pod_type=Constants.DEFAULT_POD_TYPE,
             )
-            return vector_db
-        except Exception as e:
-            raise AdapterError(str(e))
+        elif specification == Constants.SPEC_SERVERLESS:
+            cloud = self._config.get(Constants.CLOUD)
+            region = self._config.get(Constants.REGION)
+            spec = ServerlessSpec(cloud=cloud, region=region)
+        logger.info(f"Setting up Pinecone spec for {spec}")
+        try:
+            self._client.describe_index(name=self._collection_name)
+        except NotFoundException:
+            logger.info(f"Index:{self._collection_name} does not exist. Creating it.")
+            self._client.create_index(
+                name=self._collection_name,
+                dimension=dimension,
+                metric=Constants.METRIC,
+                spec=spec,
+            )
+        self.vector_db: BasePydanticVectorStore = PineconeVectorStore(
+            index_name=self._collection_name,
+            api_key=str(self._config.get(Constants.API_KEY)),
+            environment=str(self._config.get(Constants.ENVIRONMENT)),
+        )
+        return self.vector_db
 
     def test_connection(self) -> bool:
-        self.config[VectorDbConstants.EMBEDDING_DIMENSION] = (
+        self._config[VectorDbConstants.EMBEDDING_DIMENSION] = (
             VectorDbConstants.TEST_CONNECTION_EMBEDDING_SIZE
         )
         vector_db = self.get_vector_db_instance()
         test_result: bool = VectorDBHelper.test_vector_db_instance(
             vector_store=vector_db
         )
         # Delete the collection that was created for testing
-        if self.client:
-            self.client.delete_index(self.collection_name)
+        if self._client:
+            self._client.delete_index(self._collection_name)
         return test_result
+
+    def close(self, **kwargs: Any) -> None:
+        # Close connection is not defined for this client
+        pass
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: dict[Any, Any]) -> None:
+        specification = self._config.get(Constants.SPECIFICATION)
+        if specification == Constants.SPEC_SERVERLESS:
+            # To delete all records representing chunks of a single document,
```

Comment:
@gaya3-zipstack  can you give a bit more of context here. We can connect once you available

Reply:
Pinecone has 2 types of indexes
1. Serverless
2. POd-based (2 types again -> Free/Starter index and Paid index)
For serverless, we can only delete by using the ID of the record. Whereas Pod-based paid index, we can delete by using Metadata filter also. 
Llama index is not handling this limitation today. Hence we had to write a specific implementation to delete the nodes by passing in IDs